core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1114,6 +1115,10 @@ private boolean hasNestedAggregations(Aggregate rel) {
           for (AggregateCall aggregateCall : rel.getAggCallList()) {
             aggregatesArgs.addAll(aggregateCall.getArgList());
           }
+          aggregatesArgs.addAll(rel.getGroupSet().toList());
```

Comment:
Looking at `Aggregate` isn't `groupSet` always inside `groupSets`? In which case you only need to iterate over values in  `groupSets`.

Reply:
Thanks, it seems the case. Removed.